### PR TITLE
Make `core_tap_version_string` reusable.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

`brew --version` and `brew cask --version` both use the same format to output the tap version, so it makes sense to have a common method for that in the `Tap` class.

@Homebrew/cask, also updated `HBC_VERSION` to `0.60.1`, since that is the actual latest tag, but that also doesn't make sense anymore, so we should probably just use `HOMEBREW_VERSION` instead.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
> @Homebrew/cask, also updated `HBC_VERSION` to `0.60.1`, since that is the actual latest tag, but that also doesn't make sense anymore, so we should probably just use `HOMEBREW_VERSION` instead.

Fine by me.

File: Library/Homebrew/brew.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 if ARGV == %w[--version] || ARGV == %w[-v]
   puts "Homebrew #{HOMEBREW_VERSION}"
-  puts "Homebrew/homebrew-core #{Homebrew.core_tap_version_string}"
+  puts "Homebrew/homebrew-core #{CoreTap.instance.version_string}"
```
Comment (jawshooah) [inline Library/Homebrew/brew.rb]: Looks like `CoreTap` hasn't been initialized yet at this point: https://travis-ci.org/Homebrew/brew/builds/164644934#L518-L520

File: Library/Homebrew/cask/lib/hbc/version.rb [hunk 2]
```diff
@@ -1,13 +1,10 @@
-HBC_VERSION = "0.60.0".freeze
+HBC_VERSION = "0.60.1".freeze
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/lib/hbc/version.rb]: Is it worth maintaining a separate version for the two projects? Longer-term at least this should go away.

Comment (MikeMcQuaid) [inline Library/Homebrew/cask/lib/hbc/version.rb]: I see this was answered to 👍 to using Homebrew's version.

**Comment by MikeMcQuaid** [comment]
Tests are failing: https://bot.brew.sh/job/Brew%20Pull%20Requests/2263/version=yosemite/testReport/junit/brew-test-bot/yosemite/tests___no_compat/

Otherwise 👍 to the general approach.

**Comment by reitermarkus** [comment]
Should be ready to merge now.

File: Library/Homebrew/brew.rb [hunk 3]
```diff
@@ -14,8 +14,9 @@
 require "global"
 
 if ARGV == %w[--version] || ARGV == %w[-v]
+  require "tap"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/brew.rb]: Maybe consider just requiring that in `global`?

File: Library/Homebrew/cask/lib/hbc/version.rb [hunk 4]
```diff
@@ -1,13 +1,8 @@
...
 module Hbc
   def self.full_version
     @full_version ||= begin
-      revision, commit = Dir.chdir(Hbc.default_tap.path) do
-        [`git rev-parse --short=4 --verify -q HEAD 2>/dev/null`.chomp,
-         `git show -s --format="%cr" HEAD 2>/dev/null`.chomp]
-      end
-      "#{HBC_VERSION} (git revision #{revision}; last commit #{commit})"
+      "Homebrew-Cask #{HOMEBREW_VERSION}\n" \
+      "caskroom/homebrew-cask #{Hbc.default_tap.version_string}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/lib/hbc/version.rb]: Favour `EOS.undent` over embedded `\n`.

File: Library/Homebrew/tap.rb [hunk 5]
```diff
@@ -138,6 +138,16 @@ def to_s
     name
   end
 
+  def version_string
+    return "N/A" unless installed?
+    if pretty_revision = git_short_head
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: Could always make it more readable with:
```
pretty_revision = git_short_head
return "(no git repository)" unless pretty_revision
...
```

File: Library/Homebrew/brew.rb (hunk 3 above)
Comment (reitermarkus) [inline Library/Homebrew/brew.rb]: Let's do that in a separate PR, as `tap` is `require`d in many other places.

Comment (MikeMcQuaid) [inline Library/Homebrew/brew.rb]: 🆒 👍
